Add new tutorial (Beer-data-analysis) (#3483)
galaxyproject/training-material: topics/metagenomics/tutorials/beer-data-analysis/tutorial.md

---
layout: tutorial_hands_on

title: 'Identification of yeasts in a beer'

> zenodo_link: ''

Reviewer: Please add the Zenodo link here, since it is used in the upload step.

questions:
- How can yeast strains in a beer sample be identified?
- How can we process metagenomic data sequenced using Nanopore?
objectives:
- Inspect metagenomics data
- Run metagenomics tools
- Identify yeast strains contained in a sequenced beer sample using DNA
- Visualize the microbiome community of a beer sample
time_estimation: 1H
key_points:

> - TO REPHRASE

Reviewer 2: yep :smile:

shape and size and many other things unique for a bacterial species. Because of
the encoding information the DNA can be used to identify what kind of bacteria
the DNA is from. Therefore, within a sample form soil, our gut or beer one can
specify what kind of species are inside the sample.

In this tutorial, we will use data generated via the [BeerDEcoded project](https://streetscience.community/projects/beerdecoded/).

> ### {% icon comment %} The BeerDEcoded project
>
> In the BeerDEcoded project, workshops are organized with schools and general
> audience to introduce biology and genomic science and make people aware about
> DNA, sequencing technologies, bioinformatics, open science and their
> applications and their impact on their daily life.
>
> The workshops 1-2 days continuous (or divided over several days) practical workshop on beer decoding, where participants:
> 1. Extract yeasts and their DNA from beer bottle,
> 2. Sequence the extracted DNA using a MinION sequencer to obtain the sequence of bases/nucleotides (A, T, C and G) for each DNA fragment in the sample,
> 3. Analyze the sequenced data in order to know which organisms this DNA is from
>

> > ![The image represents a BeerDEcoded workshop. On the left, there is a beer tool. An arrow goes from the bottle to DNA with "Extraction" written on the below. An arrow goes from DNA to DNA sequences with "Sequencing" written on the below. An arrow goes from the DNA sequences to Yeasts with "Data analysis" written on the below](./images/beerprocess.png)

Author: yes. I changed it.

MinION to obtain sequences of DNA of the extracted yeasts. Now, we would like using
these sequences to identify to which yeast species they belong and then getting an idea
about the diversity of organisms in the beer, also called microbiome community.

To get this information, we need to process the sequence data in few steps:
1. Check the quality of the data
2. Assign taxonomic label, i.e. species, to the sequences
3. Visualize the species distribution

This type of data analysis requires running several bioinformatics tools and
usually require informatics background. [Galaxy](https://galaxyproject.org/) is
an open-source platform for data analysis that enables users to use bioinformatics
tools through its graphical web interface, accessible via any Web browser.

So, in this tutorial, we will use Galaxy to extract and visualize the community
of yeasts from a beer bottle.

> ### Agenda
>
> In this tutorial, we will cover:
>
> 1. TOC
> {:toc}
>
{: .agenda}

# Prepare Galaxy and data

First of all, this turorial will get you hands on the basic Galaxy tasks, including creating a history and importing data.

> ## Get familier with Galaxy

Reviewer: ```suggestion
## Get familiar with Galaxy
```

> ### {% icon hands_on %} Hands-on: Open Galaxy
> 1. Open your favorite browser (Chrome, Safari or Firefox as your browser, not Internet Explorer!)
> 2. Browse to a Galaxy instance: the one recommended by your instructor or one in the list **Galaxy instance** on the head of this page
> 3. Log in or register (top panel)
>

> >    ![Login or Register on the top panel](./images/login_register.png)

Author: thanks for the advice, this is done.

{: .hands_on}

The Galaxy homepage is divided into three panels:
* Tools on the left
* Viewing panel in the middle
* History of analysis and files on the right

> ![Galaxy interface screenshot showing history panel on the right, tools panel on the left, and main panel at the center](./images/galaxy_interface.png "The Galaxy interface")

Reviewer 3: thanks for writing good alt text!

> ### {% icon hands_on %} Hands-on: Prepare the Galaxy history
>
> 1. Create a new history for this analysis
>
>    {% snippet faqs/galaxy/histories_create_new.md %}
>
> 2. Rename the history
>
>    {% snippet faqs/galaxy/histories_rename.md %}
>
{: .hands_on}

## Get data

Before we can begin any Galaxy analysis, we need to upload the input data: FASTQ files

> ### {% icon hands_on %} Hands-on: Upload your dataset
>
> 1. Import the sequenced data
>
>    - Option 1 [{% icon video %}](https://youtu.be/FFCDx1rMGAQ): Your own local data using **Upload Data** (recommended for 1-10 datasets).
>
>      {% snippet faqs/galaxy/datasets_upload.md %}
>
>    - Option 2: From Zenodo, an external server via URL
>
>      ```text
>      {{ page.zenodo_link }}
>      ```
>
>      {% snippet faqs/galaxy/datasets_import_via_link.md %}
>
>    Your uploaded file is now in your current history. When the file has uploaded to Galaxy, it will turn green. But, what is this file?
>
> 2. Click on the {% icon galaxy-eye %} (eye) icon next to the dataset name, to look at the file content
>
{: .hands_on}

The contents of the file will be displayed in the central Galaxy panel.

> This file contains the sequences, also called **reads**, of DNA, *i.e.* sucession of nucleotides, for all fragments from the yeasts in the beer, in FASTQ format.

Reviewer: ```suggestion
This file contains the sequences, also called **reads**, of DNA, *i.e.* succession of nucleotides, for all fragments from the yeasts in the beer, in FASTQ format.
```

> > ### {% icon details %} FASTQ format

Reviewer: I like this explanation, and I think it would be useful in other tutorials as well, would you mind making it an FAQ? maybe in the sequence-analysis topic? then we can include it in different tutorials :)

> +
> $'())#$$%#$%%'-$&$%'%#$%('+;<>>>18.?ACLJM7E:CFIMK<=@0/.4<9<&$007:,3<IIN<3%+&$(+#$%'$#$.2@401/5=49IEE=CH.20355>-@AC@:B?7;=C4419)*$$46211075.$%..#,529,''=CFF@:<?9B522.(&%%(9:3E99<BIL?:>RB--**5,3(/.-8B>F@@=?,9'36;:87+/19BAD@=8*''&''7752'$%&,5)AM<99$%;EE;BD:=9<@=9+%$
> ```
>
> It means that the fragment named `@03dd2268-71ef-4635-8bce-a42a0439ba9a` corresponds to:
> - the DNA sequence`AGTAAGTAGCGAACCGGTTTCGTTTGGGTGTTTAACCGTTTTCGCATTTATCGTGAAACGCTTTCGCGTTTTCGTGCGGAAGGCGCTTCACCCAGGGCCTCTCATGCTTTGTCTTCCTGTTTATTCAGGATCGCCCAAAGCGAGAATCATACCACTAGACCACACGCCCGAATTATTGTTGCGTTAATAAGAAAAGCAAATATTTAAGATAGGAAGTGATTAAAGGGAATCTTCTACCAACAATATCCATTCAAATTCAGGCA` and
> - this sequence has been sequenced with a quality `$'())#$$%#$%%'-$&$%'%#$%('+;<>>>18.?ACLJM7E:CFIMK<=@0/.4<9<&$007:,3<IIN<3%+&$(+#$%'$#$.2@401/5=49IEE=CH.20355>-@AC@:B?7;=C4419)*$$46211075.$%..#,529,''=CFF@:<?9B522.(&%%(9:3E99<BIL?:>RB--**5,3(/.-8B>F@@=?,9'36;:87+/19BAD@=8*''&''7752'$%&,5)AM<99$%;EE;BD:=9<@=9+%$`.
>
> But what does this quality score mean?
>

> > The quality score for each sequence is a string of characters, one for each base of the nucleic sequence, used to characterize the probability of mis-identification of each base. The score is encoded using the ASCII character table (with [some historical differences](https://en.wikipedia.org/wiki/FASTQ_format#Encoding)):

Reviewer 4: ```suggestion
> The quality score for each sequence is a string of characters, one for each base of the nucleotide sequence, used to characterize the probability of misidentification of each base. The score is encoded using the ASCII character table (with [some historical differences](https://en.wikipedia.org/wiki/FASTQ_format#Encoding)):
```

> 60 | 1 in 1,000,000 | 99.9999%
>
{: .details }

# Data quality

## Assess data quality

Before starting to work on our data, it is necessary to assess its quality. This is an essential step if we aim to obtain a **meaningful downstream analysis**.

**FastQC** is one of the most widely used tools to **check the quality** of data generated by High Throughput Sequencing (HTS) technologies.

> ### {% icon hands_on %} Hands-on: Quality check
>
> 1. {% tool [FASTQC](toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0) %} with the following parameters
>    - {% icon param-files %} *"Raw read data from your current history"*: `Reads`
>
> 2. Inspect the generated HTML file
>
{: .hands_on}

**FastQC** provides information on various parameters, such as the range of quality values across all bases at each position:

![FastQC Per base sequence quality with scores below 200](./images/fastqc_1.png "Per base sequence quality")

We can see that the quality of our sequencing data grows after the first few bases, stays around a score of 18, which is a relatively low value compared to other sequencing technologies, and then decreases again at the end of the sequences.

> 

Reviewer: Maybe we can mention the QC tutorial here, for people who want more information about the FASTQC plots, what do you think/.

- **fastp** ({% cite Chen_2018 %}) to filter sequences with low quality score (below 9)

> ### {% icon hands_on %} Hands-on: Improve the dataset quality
>
> 1. {% tool [Porechop](toolshed.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.3) %} with the following parameters:
>    - *"Input FASTA/FASTQ"*: {% icon param-files %} `Reads`
>    - *"Output format for the reads"*: `fastq`
>
> 2. {% tool [fastp](toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0) %} with the following parameters:
>    - *"Single-end or paired reads"*: `Single-end`
>     - {% icon param-collection %} *"Dataset collection"*: output of **Porechop**
>        - In *"Adapter Trimming Options"*:
>            - *"Disable adapter trimming"*: `Yes`
>    - In *"Filter Options"*:
>        - In *"Quality filtering options"*:
>            - *"Qualified quality phred"*: `9`
>    - In *"Read Modification Options"*:
>        - *"PolyG tail trimming"*: `Disable polyG tail trimming`
>
> 3. (Optional) {% tool [FASTQC](toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0) %} with the following parameters
>    - {% icon param-files %} *"Raw read data from your current history"*: `output of **fastp**
>

> > 4. (Optional) Inspect the output of **FASTQC** to see how the quality has been improved

Reviewer 2: Added

{: .hands_on}

# Assign taxonomic classifications

One of the key steps in metagenomic data analysis is to identify the taxon to which the individual reads belong.

> ### {% icon comment %} Taxon?
>
> Taxonomy is the method used to naming, defining (circumscribing) and classifying groups of biological organisms based on shared characteristics. It is founded on the concept that morphological similarities descend from a common evolutionary ancestor.
>
> At a very broad level, all living organisms are grouped into one of the 7 Kingdoms of Life each of which are split further into Phylum > Class > Order > Family > Genus > Species.
>
> For example for the cat:
>
> Level | Classification
> --- | ---
> Kingdom | Animalia
> Phylum | Chordata
> Class | Mammalia
> Order | Carnivora
> Family | Felidae
> Genus | *Felis*
> Species |	*F. catus*
>

> > A taxon is a then group of organisms or of population(s) of organisms that form a taxonomic unit. A taxon is usually assigned a rank when it is given its formal, Latin name.

Reviewer 4: Somthing is wrong here I think:
> A taxon is a then a group of organisms or of population(s) of organisms that form a taxonomic unit. 

Maybe:
A taxon is a group of organisms and/ or populations(s) of organisms that together form a taxonomic unit.

> 2. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2) %} with the following parameters:
>    - {% icon param-file %} *"File to process"*: `out_file` (output of **Reverse** {% icon tool %})
>    - In *"Replacement"*:
>        - {% icon param-repeat %} *"Insert Replacement"*
>            - *"Find pattern"*: `\|`
>            - *"Replace with:"*: `\t`
{: .hands_on}

**Krona** creates an interactive report that allows hierarchical data (like taxonomy) to be explored with zooming, multi-layered pie charts. With this tool, we can easily visualize the composition of the microbiome communities and also compare how the populations of microorganisms are modified between different samples.

> ### {% icon hands_on %} Hands-on: Krona pie chart
>
> 1. {% tool [Krona pie chart](toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1) %} with the following parameters:
>    - *"What is the type of your input data"*: `Tabular`
>        - {% icon param-file %} *"Input file"*:  output of **Replace Text** {% icon tool %}
>
> 2. Inspect the generated file
{: .hands_on}

Let's take a look at the result.

<!--

> <iframe id="krona" src="krona_all.html" frameBorder="0" width="100%" height="900px"> ![Krona](./images/krona.png) </iframe>

Reviewer: if you include the krona html file here instead of a png it would be interactive and embedded, could be nice?

---
layout: tutorial_hands_on

title: Identification of yeasts in a beer using Nanopore sequencing

> zenodo_link: https://doi.org/10.5281/zenodo.6620778

Reviewer 2: Reupload data to Zenodo

Reviewer 4: The new Zenodo version: [10.5281/zenodo.7093173](https://zenodo.org/badge/DOI/10.5281/zenodo.7093173.svg)

3. Visualize the distribution of the different species

This type of data analysis requires running several bioinformatics tools and
usually requires a computer science background. [Galaxy](https://galaxyproject.org/) is
an open-source platform for data analysis that enables anyone to use bioinformatics
tools through its graphical web interface, accessible via any Web browser.

So, in this tutorial, we will use Galaxy to extract and visualize the community
of yeasts from a beer bottle.

> ### Agenda
>
> In this tutorial, we will cover:
>
> 1. TOC
> {:toc}
>
{: .agenda}

# Prepare Galaxy and data

First of all, this tutorial will get you hands on with some basic Galaxy tasks, including creating a history and importing data.

## Get familiar with Galaxy

> ### {% icon hands_on %} Hands-on: Open Galaxy
>
> 1. Open your favorite browser (Chrome, Safari or Firefox as your browser, not Internet Explorer!)
> 2. Create a Galaxy account if you do not have one
>
>    {% snippet faqs/galaxy/galaxy_creating_an_account.md %}
>
{: .hands_on}

The Galaxy homepage is divided into three panels:
* Tools on the left
* Viewing panel in the middle
* History of analysis and files on the right

> ![Galaxy interface screenshot showing history panel on the right, tools panel on the left, and main panel at the center](./images/galaxy_interface.png "The Galaxy interface")

Reviewer 2: I changed the image used here. We will need to update the image in the Introduction folder (outside the scope of this PR)

The first time you use Galaxy, there will be no files in your history panel.

Any analysis should get its own Galaxy history. So let's start by creating a new one:

> ### {% icon hands_on %} Hands-on: Prepare the Galaxy history
>
> 1. Create a new history for this analysis
>
>    {% snippet faqs/galaxy/histories_create_new.md %}
>
> 2. Rename the history
>
>    {% snippet faqs/galaxy/histories_rename.md %}
>
{: .hands_on}

## Get data

> Before we can begin any Galaxy analysis, we need to upload the input data: FASTQ files

Reviewer 2: We can get the FASTQ files directly from the Minion.

{% snippet topics/sequence-analysis/faqs/fastq.md %}

# Data quality

## Assess data quality

Before starting to work on our data, it is necessary to assess its quality. This is an essential step if we aim to obtain a **meaningful downstream analysis**.

**FastQC** is one of the most widely used tools to **check the quality** of data generated by High Throughput Sequencing (HTS) technologies.

> ### {% icon hands_on %} Hands-on: Quality check
>
> 1. {% tool [FASTQC](toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0) %} with the following parameters
>    - {% icon param-file %} *"Raw read data from your current history"*: `Reads`
>
> 2. Inspect the generated HTML file
>
{: .hands_on}

> ### {% icon question %} Questions
>
> Given the Basic Statistics table on the top,
> 1. How many sequences are in the FASTQ file?
> 2. How long are the sequences?
>
> > ### {% icon solution %} Solution
> >
> > 1. There are 1876 sequences.
> > 2. The sequences range from 130 nucleotides to 2327 nucleotides. Not all sequences have then the same length.
> >

> > {: .solution}

Reviewer 2: I added a screenshot of the table here

>
{: .question}

**FastQC** provides information on various parameters, such as the range of quality values across all bases at each position:

> ![FastQC Per base sequence quality with scores below 200](./images/fastqc_1.png "Per base sequence quality")

Reviewer 2: Added

> We can see that the quality of our sequencing data grows after the first few bases, stays around a score of 18, which is a relatively low value compared to other sequencing technologies, and then decreases again at the end of the sequences.

Reviewer 2: I did not want to go too much in details there given the audience but I put the paper as reference (thanks for it)

> >
> >
> > 2. The main expected microorganism is ***Saccharomyces cerevisiae*** with its taxonomy
> >
> >    Level | Classification
> >    --- | ---
> >    Domain | Eukaryota
> >    Kingdom | Fungi
> >    Phylum | Ascomycota
> >    Class | Saccharomycetes
> >    Order | Saccharomycetales
> >    Family | Saccharomycetaceae
> >    Genus | *Saccharomyces*
> >    Species | *S. cerevisiae*
> >
> {: .solution}
>
{: .question}

Taxonomic assignment or classification is the process of assigning an **Operational Taxonomic Unit** (OTUs, that is, groups of related individuals / taxon) to sequences. To assign an OTU to a sequence it is compared against a database, but this comparison can be done in different ways, with different bioinformatics tools. Here we will use **Kraken2** ({% cite wood2019improved %}).

> ### {% icon details %} Kraken2 and the k-mer approach for taxonomy classification
>
> In the $$k$$-mer approach for taxonomy classification, we use a database containing the DNA sequences of genomes we know the taxonomy. The genome database is broken into pieces of length $$k$$ (called $$k$$-mers), usually 30bp.
>

> > **Kraken** examines the $$k$$-mers within the query sequence, searches for them in the database, looks for where these are placed within the taxonomy tree inside the database and make the classification with the most probable position.  the maps $$k$$-mers to the lowest common ancestor (LCA) of all genomes known to contain the given $$k$$-mer.

Reviewer 2: It should be fixed now

>    - *"Print scientific names instead of just taxids"*: `Yes`
>    - In *"Create Report"*:
>        - *"Print a report with aggregrate counts/clade to file"*: `Yes`
>        - *"Format report output like Kraken 1's kraken-mpa-report"*: `Yes`
>    - *"Select a Kraken2 database"*: `Prebuilt Refseq indexes: PlusPF`
>
>      The database here contains reference sequences and taxonomies. We need to
>      be sure it contains yeasts, i.e. fungi.
>
> 2. Inspect the report file
{: .hands_on}

The Kraken report in MPA format is a tabular file with 2 columns:

```
d__Eukaryota	756
d__Eukaryota|k__Metazoa	402
d__Eukaryota|k__Metazoa|p__Chordata	402
d__Eukaryota|k__Metazoa|p__Chordata|c__Mammalia	402
d__Eukaryota|k__Metazoa|p__Chordata|c__Mammalia|o__Primates	402
d__Eukaryota|k__Metazoa|p__Chordata|c__Mammalia|o__Primates|f__Hominidae	402
d__Eukaryota|k__Metazoa|p__Chordata|c__Mammalia|o__Primates|f__Hominidae|g__Homo	402
d__Eukaryota|k__Metazoa|p__Chordata|c__Mammalia|o__Primates|f__Hominidae|g__Homo|s__Homo sapiens	402
d__Eukaryota|k__Fungi	342
d__Eukaryota|k__Fungi|p__Ascomycota	341
d__Eukaryota|k__Fungi|p__Ascomycota|c__Saccharomycetes	336

> ...

Reviewer 2: It was to show that it is the just the beginning of the file

> Beer is alive! It contains microorganisms, in particular **yeasts**.
>
> Indeed, grain and water create a sugary liquid (called wort). The beer brewer

> > adds yeasts to it. By eating the sugar, yeasts creates alcohol.

Reviewer 5: It should be a comma, not a dot I think.
